argh/tests/help_json_tests.rs

```diff
@@ -0,0 +1,640 @@
+"description": "Top-level command.",
+"options": [{"short": "", "long": "--help", "description": "display usage information"},
+{"short": "", "long": "--help-json", "description": "display usage information encoded in JSON"}],
+"positional": [],
+"examples": "",
+"notes": "",
+"error_codes": [],
+"subcommands": [{"name": "one", "description": "First subcommand."},
+{"name": "two", "description": "Second subcommand."}]
+}
+"###,
+    );
+
+    assert_help_json_string::<TopLevel>(
+        vec!["one", "--help-json"],
+        r###"{
+"usage": "test_arg_0 one --x <x>",
+"description": "First subcommand.",
+"options": [{"short": "", "long": "--x", "description": "how many x"},
+{"short": "", "long": "--help", "description": "display usage information"},
+{"short": "", "long": "--help-json", "description": "display usage information encoded in JSON"}],
+"positional": [],
+"examples": "",
+"notes": "",
+"error_codes": [],
+"subcommands": []
+}
+"###,
+    );
+}
+
+#[test]
+fn help_json_test_multiline_doc_comment() {
+    #[derive(FromArgs)]
+    /// Short description
+    struct Cmd {
+        #[argh(switch)]
+        /// a switch with a description
+        /// that is spread across
+        /// a number of
+        /// lines of comments.
+        _s: bool,
+    }
+    assert_help_json_string::<Cmd>(
+        vec!["--help-json"],
+        r###"{
+"usage": "test_arg_0 [--s]",
+"description": "Short description",
+"options": [{"short": "", "long": "--s", "description": "a switch with a description that is spread across a number of lines of comments."},
+{"short": "", "long": "--help", "description": "display usage information"},
+{"short": "", "long": "--help-json", "description": "display usage information encoded in JSON"}],
+"positional": [],
+"examples": "",
+"notes": "",
+"error_codes": [],
+"subcommands": []
+}
+"###,
+    );
+}
+
+#[test]
+fn help_json_test_basic_args() {
+    #[allow(dead_code)]
+    #[derive(FromArgs)]
+    /// Basic command args demonstrating multiple types and cardinality. "With quotes"
+    struct Basic {
+        /// should the power be on. "Quoted value" should work too.
+        #[argh(switch)]
+        power: bool,
+
+        /// option that is required because of no default and not Option<>.
+        #[argh(option, long = "required")]
+        required_flag: String,
+
+        /// optional speed if not specified it is None.
+        #[argh(option, short = 's')]
+        speed: Option<u8>,
+
+        /// repeatable option.
+        #[argh(option, arg_name = "url")]
+        link: Vec<String>,
+    }
+    assert_help_json_string::<Basic>(
+        vec!["--help-json"],
+        r###"{
+"usage": "test_arg_0 [--power] --required <required> [-s <speed>] [--link <url...>]",
+"description": "Basic command args demonstrating multiple types and cardinality. \"With quotes\"",
+"options": [{"short": "", "long": "--power", "description": "should the power be on. \"Quoted value\" should work too."},
```

Comment:
nit: I know argh uses the term `option` but to me it's a little ambiguous to say "required option". Option sounds too much like "optional". You may feel differently, and this is really in the weeds, but if your schema is not too rigid, maybe consider calling the JSON array "args" or "flags"?

@erickt feel free to weigh in if you think I'm off base.